extensions/cornerstone/src/Viewport/Overlays/CustomizableViewportOverlay.tsx

```diff
@@ -0,0 +1,532 @@
+      {
+        id: 'WindowLevel',
+        customizationType: 'ohif.overlayItem.windowLevel',
+      },
+    ];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`topLeftOverlayItem_${i}`}>{_renderOverlayItem(item)}</div>
+        ))}
+      </>
+    );
+  }, [
+    topLeftCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
+    customizationService,
+    instance,
+    voi,
+    scale,
+    instanceNumber,
+  ]);
+
+  const getTopRightContent = useCallback(() => {
+    const items = topRightCustomization?.items || [
+      {
+        id: 'InstanceNmber',
+        customizationType: 'ohif.overlayItem.instanceNumber',
+      },
+    ];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`topRightOverlayItem_${i}`}>{_renderOverlayItem(item)}</div>
+        ))}
+      </>
+    );
+  }, [
+    topRightCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
+    customizationService,
+    instance,
+    voi,
+    scale,
+    instanceNumber,
+  ]);
+
+  const getBottomLeftContent = useCallback(() => {
+    const items = bottomLeftCustomization?.items || [];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`bottomLeftOverlayItem_${i}`}>
+            {_renderOverlayItem(item)}
+          </div>
+        ))}
+      </>
+    );
+  }, [
+    bottomLeftCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
+    customizationService,
+    instance,
+    voi,
+    scale,
+    instanceNumber,
+  ]);
+
+  const getBottomRightContent = useCallback(() => {
+    const items = bottomRightCustomization?.items || [];
+    return (
+      <>
+        {items.map((item, i) => (
+          <div key={`bottomRightOverlayItem_${i}`}>
+            {_renderOverlayItem(item)}
+          </div>
+        ))}
+      </>
+    );
+  }, [
+    bottomRightCustomization,
+    element,
+    viewportData,
+    imageSliceData,
+    viewportIndex,
+    servicesManager,
+    customizationService,
+    instance,
+    voi,
+    scale,
+    instanceNumber,
+  ]);
+
+  return (
+    <ViewportOverlay
+      topLeft={getTopLeftContent()}
+      topRight={getTopRightContent()}
+      bottomLeft={getBottomLeftContent()}
+      bottomRight={getBottomRightContent()}
+    />
+  );
+}
+
+function _getViewportInstance(viewportData, imageIndex) {
+  let imageId = null;
+  if (viewportData.viewportType === Enums.ViewportType.STACK) {
+    imageId = viewportData.data.imageIds[imageIndex];
+  } else if (viewportData.viewportType === Enums.ViewportType.ORTHOGRAPHIC) {
+    const volumes = viewportData.volumes;
+    if (volumes && volumes.length == 1) {
+      const volume = volumes[0];
+      imageId = volume.imageIds[imageIndex];
+    }
+  }
+  return imageId ? metaData.get('instance', imageId) || {} : {};
+}
```

Comment:
This can be re-written via csViewports (get them from cornerstoneViewportService.getCornerstone...) and then you can say

`viewport.getCurrentImageId()`

Reply:
I remember when I implemented this, the `getCurrentImageId()` didn't work well

Reply:
just tried briefly, and it doesn't return the `imageId` correctly. we need to investigate furthure